docs/expressions.md

```diff
@@ -176,6 +178,68 @@ lstat returns various summary stats per bucket for the specified `field`. The fi
   * As of January 15, 2015 - logstash functionality is new so these functions may change a fair amount based on experience using them in alerts.
   * Alerts using this information likely want to set ignoreUnknown, since only "groups" that appear in the time frame are in the results.
 
+## Elastic Query Functions
+
+Elasitc replaces the deprecated logstash (ls) functions. It only works with Elastic v2+. It introduces two new types to allow for greater flexibility in querying. The ESIndexer type generates indexes to query (based on the date range). There are now different functions to generate indexers for people with different configurations. The ESQuery type is generates elastic queries so you can filter your results. By making these new types, new Indexers and Elastic queries can be added over time.
```

Comment:
@Dieterbe  Right, this meant to make it so we don't tie to a particular schema. There are elastic queries beyond what I created functions for, but if people request them they can just be new funcs that return ESQuery. Same goes for the index naming schema. 

Regexes might not all make it through the parser - there is an open issue for that. But I think that would be fixed independently of this.

One limitation I will add is that if your fields is analyzed, you get multiple histograms based on the terms. But turning an analyzed field into a tag key doesn't really make sense in any scenario I can think of.